src/commands/createFunction/promptStepsV2/PromptSchemaBaseStep.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from "@microsoft/vscode-azext-utils";
+import { localize } from "../../../localize";
+import { ParsedInput } from "../../../templates/script/parseScriptTemplatesV2";
+import { FunctionV2WizardContext } from "../FunctionV2WizardContext";
+
+// TODO: Create more prompt steps where value is used to determine the prompt (string = input, boolean/enum = quickpick)
+// if required is false, then add a skip for now (for quickpick only) or allow "undefined"
+export abstract class PromptSchemaBaseStep<T extends FunctionV2WizardContext> extends AzureWizardPromptStep<T> {
+
+    public constructor(readonly input: ParsedInput) {
+        super();
+    }
+
+    public async prompt(context: T): Promise<void> {
+        context[this.input.assignTo] = await this.promptAction(context);
+    }
+
+    protected abstract promptAction(context: T): Promise<unknown>;
+
+    protected validateInput(input: string | undefined): string | undefined {
+        if (!input && this.input.required) {
+            return localize('promptV2StepEmpty', 'The input cannot be empty.');
+        }
+
+        const validators = this.input.validators || [];
+        for (const validator of validators) {
+            if (input) {
+                if (new RegExp(validator.expression).test(input)) {
+                    return validator.errorText;
+                }
+            }
+        }
+
+        // special case for existingfile/new file
+
+        return undefined;
+    }
+
+    public shouldPrompt(context: T): boolean {
+        return !context[this.input.assignTo];
```

Comment:
If the input type is boolean, and the user inputs `false` and `context[this.input.assignTo] === false` then `shouldPrompt()` will return `true` when it should return `false` since the user has already completed this prompt.

Certainly an edge case, but I could see it causing issues.

We probably should just check if the value is not null or undefined.